### ISSUE TITLE
Ability to use short style in newer Dart versions

### ISSUE THREAD

**Issue by Reprevise** [dart-lang/dart_style#1684]
While there are some decent changes with the new tall style, I can't bring myself to use it at the moment due to some open issues here. That means I also can't use new Flutter/Dart releases (w/o manually formatting my changes, which I also can't be bothered to do). Is there a way that a `style` parameter can be passed to `dart format`, either `tall` or `short`? At least until support for the short style is removed (and hopefully the big outstanding issues are fixed by then).

**Comment by eernstg** [comment]
Perhaps `dart format format --language-version=3.6 ...`?

**Comment by Reprevise** [comment]
Would it throw an error if it encounters new language features like wildcard variables, etc?

**Comment by eernstg** [comment]
I think the first new language feature which is detectable by a parser (that is, the syntax didn't exist in earlier versions) is the null-aware elements of Dart 3.8 (for example, `_` is not new syntax, we just stop reporting some name clashes, and a name clash is not a grammar based property). So you should be able to stay in Dart 3.6 with respect to formatting for a while.

Also, https://github.com/dart-lang/dart_style/pull/1672 introduces the option to preserve trailing commas, and that seems to be the biggest outstanding issue at this time.

PS: @munificent would know all the details, I'm just commenting based on what I happen to know.

**Comment by munificent** [comment]
> While there are some decent changes with the new tall style, I can't bring myself to use it at the moment due to some open issues here. 

Which ones in particular? My hope is to get most of the pressing ones fixed soon and out in the next Dart SDK release.

> Is there a way that a `style` parameter can be passed to `dart format`, either `tall` or `short`? At least until support for the short style is removed (and hopefully the big outstanding issues are fixed by then).

No, there's no support for that and no plan to support it. We don't want to have to implement formatting support for every new language feature twice, test it twice, fix bugs twice, etc.

**Comment by Reprevise** [comment]
> Which ones in particular? My hope is to get most of the pressing ones fixed soon and out in the next Dart SDK release.

#1545, ternaries (https://github.com/dart-lang/dart_style/issues/1660#issuecomment-2722956285), https://github.com/dart-lang/dart_style/issues/1557#issuecomment-2752140373.

I'm not sure if this one has an issue, but this is just unnecessary.
Before
```dart
typedef DataViewBuilder<T> = Widget Function(
  BuildContext context,
  PagingState<int, T> state,
  NextPageCallback fetchNextPage,
);
```

After
```dart
typedef DataViewBuilder<T> =
    Widget Function(
      BuildContext context,
      PagingState<int, T> state,
      NextPageCallback fetchNextPage,
    );
```

**Comment by munificent** [comment]
> > Which ones in particular? My hope is to get most of the pressing ones fixed soon and out in the next Dart SDK release.
> 
> [#1545](https://github.com/dart-lang/dart_style/issues/1545),

That one is tricky, but now that #1687 is in, I might be able to make some progress on it.

> ternaries ([#1660 (comment)](https://github.com/dart-lang/dart_style/issues/1660#issuecomment-2722956285)),

That's #1465 and is fixed by #1687.

> [#1557 (comment)](https://github.com/dart-lang/dart_style/issues/1557#issuecomment-2752140373).

Replied on that issue.

> I'm not sure if this one has an issue, but this is just unnecessary. Before
> 
> typedef DataViewBuilder<T> = Widget Function(
>   BuildContext context,
>   PagingState<int, T> state,
>   NextPageCallback fetchNextPage,
> );
> After
> 
> typedef DataViewBuilder<T> =
>     Widget Function(
>       BuildContext context,
>       PagingState<int, T> state,
>       NextPageCallback fetchNextPage,
>     );

Probably doable. I filed #1693 to track this.

> Lack of indenting for nested ternaries:
> 
> Before:
> 
> !hasStarted
>     ? const Icon(
>       Icons.hourglass_empty,
>       // ...
>     )
>     : hasData
>         ? const Icon(/* ... */)
>         : const Icon(/* ... */),
> !hasStarted
>     ? const Icon(
>       Icons.hourglass_empty,
>       // ...
>     )
>     : hasData
>     ? const Icon(/* ... */)
>     : const Icon(/* ... */),

The missing indentation in the `Icons.hourglass_empty,` and `// ...` lines is fixed by #1626. The flattened indentation is deliberate. I know there's a longer comment about it in the issue tracker somewhere, but I can't find it.

Style-wise, the idea is that it mirrors the flattened nesting of an else-if statement chain. Importantly, it also avoids a combinatorial performance problem if the conditional expressions are all formatted separately.

**Comment by munificent** [comment]
I'm going to go ahead and close this because I think the style issues are mostly tracked elsewhere. We don't intend to support the old style in later language versions. We don't want to have to implement formatting for every new language feature multiple times.
